xdg-desktop-portal-termfilechooser: init at 0.4.0

body20002 opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
Added [xdg-desktop-portal-termfilechooser](https://github.com/boydaihungst/xdg-desktop-portal-termfilechooser) package for choosing files with terminal file chooser like [yazi](https://github.com/sxyazi/yazi)

not sure tbh how to test a systemd service but the binary compiles and seems to working
## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

body20002 committed:
xdg-desktop-portal-termfilechooser: init at 0.4.0

body20002 committed:
xdg-desktop-portal-termfilechooser: format

Eddio0141 commented:
It seems like https://github.com/boydaihungst/xdg-desktop-portal-termfilechooser has been deprecated a week ago, and they recommend to use another fork
Out of the forks, there is https://github.com/hunkyburrito/xdg-desktop-portal-termfilechooser which is active

MartinLoeper commented:
Has anyone had success to run this with wayland and XDHP?
I added it as follows but it just keeps calling the gtk file chooser:

```nix
{ pkgs, ... }:
let
  termfilechooser = (pkgs.callPackage ./pkgs/xdg-desktop-portal-termfilechooser.nix { });
in
{
  xdg = {
    portal = {
      enable = true;
      xdgOpenUsePortal = true;
      config = {
        common.default = [ "gtk" ];
        hyprland.default = [ "gtk" "hyprland" ];
        hyprland."org.freedesktop.impl.portal.FileChooser" = [ "xdg-desktop-portal-termfilechooser" ];
      };
      extraPortals = [
        pkgs.xdg-desktop-portal-gtk
        pkgs.xdg-desktop-portal-hyprland
        termfilechooser
      ];
    };
  };

  environment.systemPackages = [
    termfilechooser
  ];
}
```

body20002 commented:
> It seems like https://github.com/boydaihungst/xdg-desktop-portal-termfilechooser has been deprecated a week ago, and they recommend to use another fork Out of the forks, there is https://github.com/hunkyburrito/xdg-desktop-portal-termfilechooser which is active

I will check them.


> Has anyone had success to run this with wayland and XDHP? I added it as follows but it just keeps calling the gtk file chooser:
> 
> ```nix
> { pkgs, ... }:
> let
>   termfilechooser = (pkgs.callPackage ./pkgs/xdg-desktop-portal-termfilechooser.nix { });
> in
> {
>   xdg = {
>     portal = {
>       enable = true;
>       xdgOpenUsePortal = true;
>       config = {
>         common.default = [ "gtk" ];
>         hyprland.default = [ "gtk" "hyprland" ];
>         hyprland."org.freedesktop.impl.portal.FileChooser" = [ "xdg-desktop-portal-termfilechooser" ];
>       };
>       extraPortals = [
>         pkgs.xdg-desktop-portal-gtk
>         pkgs.xdg-desktop-portal-hyprland
>         termfilechooser
>       ];
>     };
>   };
> 
>   environment.systemPackages = [
>     termfilechooser
>   ];
> }
> ```

did you config termfilechooser too?
you need to do it in the $XDG_CONFIG_HOME/xdg-desktop-portal-termfilechooser/config
something like 
```
[filechooser]
cmd=<PATH_TO_WRAPPER>
default_dir=$HOME
```

body20002 committed:
Merge branch 'NixOS:master' into add-xdg-desktop-portal-termfilechooser

body20002 committed:
xdg-desktop-portal-termfilechooser: changed source package

body20002 commented:
I've changed the source package to https://github.com/hunkyburrito/xdg-desktop-portal-termfilechooser and removed some dependencies. I've tested the package with the new changes, and it seems to work.

FYI, if you try `GTK_USE_PORTAL=1 zenity --file-selection`, it might not work—I'm not sure why. However, trying a file picker in a browser works as intended.

Eddio0141 commented:
Very nice, I've tested this and it works perfectly
Also zenity isn't using the correct file picker for you because you should be using `GDK_DEBUG=portals` instead, but the readme seemed to have forgotten to update that example
It does mention this change at the bottom of the readme though

> Since [this merge request in GNOME](https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/4829), GTK_USE_PORTAL=1 seems to be replaced with GDK_DEBUG=portals.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/5205

michaelBelsanti commented:
> not sure tbh how to test a systemd service but the binary compiles and seems to working

Just ensure that it works as expected when added to xdg.portal.extraPortals. That will handle using the systemd service.

SuperSandro2000 commented on pkgs/by-name/xd/xdg-desktop-portal-termfilechooser/package.nix:
```diff
@@ -0,0 +1,46 @@
+  pname = "xdg-desktop-portal-termfilechooser";
+  version = "0.4.0";
+
+  src = fetchgit {
```
This must use fetchFromGitHub

body20002 committed:
refactor: use `fetchFromGitHub` instead of `fetchgit`

body20002 committed:
Merge branch 'NixOS:master' into add-xdg-desktop-portal-termfilechooser

body20002 commented:
> > not sure tbh how to test a systemd service but the binary compiles and seems to working
> 
> Just ensure that it works as expected when added to xdg.portal.extraPortals. That will handle using the systemd service.

It does work with `xdg.portal.extraPortals`.

body20002 commented:
@Anninzy it's probably something with your `yazi-wrapper.sh` try use the default one
also you shouldn't really run it like this, instead it should be enabled using systemd.
check your `systemctl --user status xdg-desktop-portal-termfilechooser.service`

body20002 commented:
@Anninzy try first activate the service, if it doesn't work try `yazi-wrapper.sh` in the `/nix/store/{path-to-termfilechooser}/share/xdg-desktop-portal-termfilechooser`.
You need to have a `config` in `~/.config/xdg-desktop-portal-termfilechooser` and `cmd` that's points to the correct wrapper you need.

you can take the easy route and just 
```bash
rm -rf ~/.config/xdg-desktop-portal-termfilechooser
cp -rv /nix/store/yvd2639qypkkdjkl3c2j8qg6cyfm5mps-xdg-desktop-portal-termfilechooser-0.4.0/share/xdg-desktop-portal-termfilechooser ~/.config
systemctl --user restart xdg-desktop-portal-termfilechooser.service
```

body20002 commented:
@Anninzy there's a problem on your end and I can't help you with it.
You should define the config & wrapper and activate the service manually and everything should be working fine.
I'm daily driving this package as my main file-chooser.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/how-to-install-xdg-desktop-portal-termfilechooser/62819/5
